src/mixins/java/org/spongepowered/common/mixin/core/world/entity/ai/goal/RunAroundLikeCrazyGoalMixin.java

```diff
@@ -45,51 +46,41 @@
 public abstract class RunAroundLikeCrazyGoalMixin extends GoalMixin {
 
     // @formatter:off
-    @Shadow @Final @Mutable private AbstractHorse horse;
+    @Shadow @Final private AbstractHorse horse;
     // @formatter:on
 
-    /**
-     * @author rexbut - December 16th, 2016
-     * @author i509VCB - February 18th, 2020 - 1.14.4
-     *
-     * @reason - adjusted to support {@link DismountTypes}
-     */
-    @Overwrite
-    public void tick() {
-        if (!this.horse.isTamed() && this.horse.getRandom().nextInt(50) == 0) {
-            Entity entity = this.horse.getPassengers().get(0);
-
-            if (entity == null) {
-                return;
-            }
-
-            if (entity instanceof Player) {
-                int i = this.horse.getTemper();
-                int j = this.horse.getMaxTemper();
-
-                if (j > 0 && this.horse.getRandom().nextInt(j) < i) {
-                    // Sponge start - Fire Tame Entity event
-                    try (CauseStackManager.StackFrame frame = PhaseTracker.getCauseStackManager().pushCauseFrame()) {
-                        frame.pushCause(entity);
-                        if (SpongeCommon.post(SpongeEventFactory.createTameEntityEvent(frame.currentCause(), (HorseLike) this.horse))) {
-                            return;
-                        }
-                    }
-                    // Sponge end
-                    this.horse.tameWithName((Player)entity);
-                    return;
-                }
-
-                this.horse.modifyTemper(5);
+    @Inject(
+        method = "tick",
+        at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/world/entity/animal/horse/AbstractHorse;tameWithName(Lnet/minecraft/world/entity/player/Player;)Z"
+        ),
+        cancellable = true,
+        locals = LocalCapture.CAPTURE_FAILHARD
+    )
+    private void impl$throwTameEntityEvent(CallbackInfo ci, Entity entity) {
```

Comment:
```suggestion
    private void impl$throwTameEntityEvent(final CallbackInfo ci, final Entity entity) {
```
We tend to use final where applicable. (There are two others)